flow/raster_cache_util.h

```diff
@@ -54,6 +54,35 @@ struct RasterCacheUtil {
     ctm.mapRect(&device_rect, rect);
     return device_rect;
   }
+
+  static SkRect GetRoundedOutDeviceBounds(const SkRect& rect,
+                                          const SkMatrix& ctm) {
+    SkRect device_rect;
+    ctm.mapRect(&device_rect, rect);
+    device_rect.roundOut(&device_rect);
+    return device_rect;
+  }
+
+  /**
+   * @brief Snap the translation components of the matrix to integers.
+   *
+   * The snapping will only happen if the matrix only has scale and translation
+   * transformations.
+   *
+   * @param ctm the current transformation matrix.
+   * @return SkMatrix the snapped transformation matrix.
+   */
```

Comment:
Consider documenting why we do this so next time we go to remove it we know what ot test.

Reply:
Done